src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,401 @@
+                    return false;
+                }
+            }
+
+            private bool EmitBindLogic(
+                ITypeSymbol type,
+                string expressionForPropertyAccess,
+                TargetMemberKind targetKind = TargetMemberKind.RootObject,
+                string? configKey = null,
+                string configVarName = LiteralFor_configuration)
+            {
+                if (UseStringBasedParse(type))
+                {
+                    string expressionForConfigValue = $@"{configVarName}[""{configKey}""]";
+                    EmitStringBasedBindLogicForType(type, expressionForPropertyAccess, expressionForConfigValue);
+                }
+                else
+                {
+                    string expressionForConfigValue = configVarName + (targetKind == TargetMemberKind.RootObject
+                        ? ""
+                        : $@".{LiteralFor_GetSection}(""{configKey}"")");
+
+                    if (type.IsValueType)
+                    {
+                        Emit_NotSupportedException_UnableToBindType(type, NotSupportedReason_ValueTypesNotSupported);
+                        return false;
+                    }
+
+                    _writer.WriteSource($@"{LiteralFor_Bind}({expressionForConfigValue}, {expressionForPropertyAccess});");
+                    _privateBindMethodGen_QueuedTypes.Enqueue(type);
+                }
+
+                return true;
+            }
+
+            private void EmitStringBasedBindLogicForType(
+                ITypeSymbol type,
+                string expressionForPropertyAccess,
+                string expressionForConfigValue)
+            {
+                string typeDisplayString = type.ToDisplayString();
+                string stringValueVarName = $"{LiteralFor_stringValue}{_stringValueCount++}";
+
+                if (type.SpecialType != SpecialType.None)
+                {
+                    string rhs = type.SpecialType switch
+                    {
+                        SpecialType.System_String => stringValueVarName,
+                        SpecialType.System_Object => DefaultKeyword,
+                        _ => $"{typeDisplayString}.Parse({stringValueVarName})"
+                    };
+                    EmitAssignment_WithNullCheck(expressionForPropertyAccess, rhs);
+                }
+                else
+                {
+                    Debug.Assert(TypeIsEnum(type));
+                    string rhs = $"({typeDisplayString}){GlobalNameForEnumType}.{LiteralFor_Parse}(typeof({typeDisplayString}), {stringValueVarName})";
+                    EmitAssignment_WithNullCheck(expressionForPropertyAccess, rhs);
+                }
+
+                void EmitAssignment_WithNullCheck(string lhs, string rhs)
+                {
+                    _writer.WriteSource($"{lhs} = {expressionForConfigValue} is {GlobalNameForStringType} {stringValueVarName} ? {rhs} : {DefaultKeyword};");
```

Comment:
Do we really want to assign the lhs with `default` when the expression evaluates to null? Sounds like we might be resetting properties in cases where no configuration is specified. Does the reflection binder do the same?